kusto/kql/query_parameters_test.go

```diff
@@ -0,0 +1,88 @@
+package kql
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/require"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestToDeclarationString(t *testing.T) {
+	tests := []struct {
+		name     string
+		b        Builder
+		qp       *StatementQueryParameters
+		failFlag bool
+		expected []string
+	}{
+		{"Test empty", NewStatementBuilder(""), NewStatementQueryParameters(), false, []string{"\n"}},
+		{"Test single add", NewStatementBuilder(""), NewStatementQueryParameters().AddString("foo", "bar"), false, []string{"declare", "query_parameters(", "foo:string", ");\n"}},
+		{"Test parameters standard",
+			NewStatementBuilder("database(databaseName).table(tableName) | where column == txt ;"),
+			NewStatementQueryParameters().
+				AddString("databaseName", "foo_1").
+				AddString("tableName", "_bar").
+				AddString("txt", "txt_"),
+			false,
+			[]string{"declare", "query_parameters(", "databaseName:string", "tableName:string", "txt:string", ");\ndatabase(databaseName).table(tableName) | where column == txt ;"}},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			q := test.b.String()
+			params := test.qp.ToDeclarationString()
+			actual := fmt.Sprintf("%s\n%s", params, q)
+			for _, sub := range test.expected {
+				require.True(t, strings.Contains(actual, sub))
+			}
+		})
+	}
+}
+func TestToParameterCollection(t *testing.T) {
```

Comment:
Do one set of tests, for each check BOTH the declaration string and the parameter collection